### PR TITLE
Fix user settings leaking into other tests

### PR DESCRIPTION
**Pull request by lukpueh**
On some platforms the user settings created in test_user_settings leak into other tests and break them, as they expect the defaults from settings.py and a clean environment.

This change modifies the user_settings testcase to restore/reset the test settings in the testcase's teardown method.

### CODE REVIEW

File: test/test_user_settings.py [hunk 1]
```diff
@@ -44,11 +49,21 @@ def setUpClass(self):
...
 
+    # Other unittests might depend on defaults:
+    # Restore monkey patched settings ...
+    for key, val in self.settings_backup.iteritems():
+      setattr(in_toto.settings, key, val)
+
+    # ... and delete test environment variables
+    del os.environ["IN_TOTO_ARTIFACT_EXCLUDE_PATTERNS"]
+    del os.environ["IN_TOTO_ARTIFACT_BASE_PATH"]
+    del os.environ["IN_TOTO_not_whitelisted"]
```
Comment (SantiagoTorres) [inline test/test_user_settings.py:64]: The rest of this variable is lowercase. is this intentional?

File: test/test_user_settings.py [hunk 2]
```diff
@@ -44,11 +49,21 @@ def setUpClass(self):
...
+    del os.environ["NOT_PARSED"]
+
```
Comment (SantiagoTorres) [inline test/test_user_settings.py:66]: This is good by me, I wonder if we can move this to a decorator for the tests. Something like

```python
@set_environ(IN_TOTO_ARTIFACT_BASE_PATH="thisthing")
def test_this_and_that(self):
  ...
```

It's probably worth doing down the line, but I'd move it to a ticket.

**Review by SantiagoTorres — APPROVED** [review]

File: test/test_user_settings.py (hunk 2 above)
Comment (lukpueh) [reply test/test_user_settings.py:66]: Oh I wasn't aware of that decorator. I can do that right away.

File: test/test_user_settings.py (hunk 1 above)
Comment (lukpueh) [reply test/test_user_settings.py:64]: Yes, to acknowledge case sensitivity in `user_settings.get_env`.

File: test/test_user_settings.py (hunk 2 above)
Comment (SantiagoTorres) [reply test/test_user_settings.py:66]: no, we need to write it

Comment (lukpueh) [reply test/test_user_settings.py:66]: oh ok :) then I guess the current approach is fine.

File: test/test_user_settings.py (hunk 1 above)
Comment (SantiagoTorres) [reply test/test_user_settings.py:64]: ah, ok ok

**Comment by SantiagoTorres** [comment]
LGTM. Merging
